src/test/java/com/mongodb/kafka/connect/sink/StartedMongoSinkTaskTest.java

```diff
@@ -0,0 +1,432 @@
+        Records::match);
+  }
+
+  @Test
+  void putTolerateAllOrderedWriteError() {
+    properties.put(MongoSinkTopicConfig.ERRORS_TOLERANCE_CONFIG, ErrorTolerance.ALL.value());
+    MongoSinkConfig config = new MongoSinkConfig(properties);
+    client.configureCapturing(
+        DEFAULT_NAMESPACE,
+        collection ->
+            when(collection.bulkWrite(anyList(), any(BulkWriteOptions.class)))
+                // batch1
+                .thenReturn(BulkWriteResult.unacknowledged())
+                // batch2
+                .thenThrow(bulkWriteException(singletonList(2), false))
+                // batch2
+                .thenReturn(BulkWriteResult.unacknowledged()));
+    StartedMongoSinkTask task =
+        new StartedMongoSinkTask(config, client.mongoClient(), errorReporter);
+    RecordsAndExpectations recordsAndExpectations =
+        new RecordsAndExpectations(
+            asList(
+                // batch1
+                Records.simpleValid(TEST_TOPIC, 0),
+                // batch2
+                Records.simpleValid(TEST_TOPIC2, 1),
+                Records.simpleValid(TEST_TOPIC2, 2),
+                Records.simpleValid(TEST_TOPIC2, 3),
+                // batch3
+                Records.simpleValid(TEST_TOPIC, 4)),
+            asList(0, 1, 2, 3, 4),
+            emptyList());
+    task.put(recordsAndExpectations.records());
+    recordsAndExpectations.assertExpectations(
+        client.capturedBulkWrites().get(DEFAULT_NAMESPACE),
+        errorReporter.reported(),
+        Records::match);
+  }
+
+  @SuppressWarnings("unchecked")
+  private static <T> T cast(final Object o) {
+    return (T) o;
+  }
+
+  private static MongoBulkWriteException bulkWriteException(
+      final List<Integer> writeErrorIndices, final boolean writeConcernError) {
+    List<BulkWriteError> writeErrors =
+        writeErrorIndices.stream()
+            .map(idx -> new BulkWriteError(0, "", BsonDocument.parse("{}"), idx))
+            .collect(Collectors.toList());
+    return new MongoBulkWriteException(
+        BulkWriteResult.unacknowledged(),
+        writeErrors,
+        writeConcernError ? new WriteConcernError(0, "", "", BsonDocument.parse("{}")) : null,
+        new ServerAddress(),
+        emptySet());
+  }
+
+  private static final class Records {
+    static SinkRecord simpleValid(final String topicName, final int idx) {
+      return new SinkRecord(
+          topicName,
+          0,
+          Schema.STRING_SCHEMA,
+          "",
+          Schema.STRING_SCHEMA,
+          String.format("{\"_id\": %d}", idx),
+          idx);
+    }
+
+    static SinkRecord simpleInvalid(final String topicName, final int idx) {
+      return new SinkRecord(
+          topicName, 0, Schema.STRING_SCHEMA, "", Schema.STRING_SCHEMA, "notJSON_" + idx, idx);
+    }
+
+    static boolean match(
+        final SinkRecord expected, final WriteModel<? extends BsonDocument> actual) {
+      ReplaceOneModel<BsonDocument> writeModel = cast(actual);
```

Comment:
I added `assertTrue(actual instanceof ReplaceOneModel)` to make the expectation clear. If for some reason in the future this method encounters a write model different from `ReplaceOneModel`, I want it to fail as early as possible (it will need to be updated to support comparing that write model), rather than returning `false` and causing a failure in a different place.